### PR TITLE
alloy: add alloy5 and take maintainership

### PR DESCRIPTION
**Pull request by bandali**
Add alloy5 (currently, 5.0.0.1), update homepage and downloadPage, and
take maintainership of the package.

###### Motivation for this change

I've started using `alloy` for my research, and noticed that the package was orphaned a while ago, and that alloy 5 wasn't available on nixpkgs. So I thought I'd change that :slightly_smiling_face: 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is my first ever contribution to nixpkgs, please feel free to point out any mistakes and/or areas of improvement.

### CODE REVIEW

File: pkgs/development/tools/alloy/default.nix [hunk 1]
```diff
@@ -1,58 +1,78 @@
...
-    install -Dm755 alloy $out/bin/alloy
+      cat << EOF > ${nameMajor}
+      #!${stdenv.shell}
+      exec ${jre}/bin/java -jar $jar "\''${@}"
+      EOF
 
-    install -Dm644 ${./icon.png} $out/share/pixmaps/alloy.png
-    cp -r ${desktopItem}/share/applications $out/share
-  '';
+      install -Dm755 ${nameMajor} $out/bin/${nameMajor}
```
Comment (Mic92) [inline pkgs/development/tools/alloy/default.nix]: Use `makeWrapper` here as described in the java chapter of the nixpkgs manual.

File: pkgs/development/tools/alloy/default.nix [hunk 2]
```diff
@@ -1,58 +1,78 @@
...
+      comment = meta.description;
+      categories = "Development;IDE;Education;";
+    };
 
-  phases = [ "installPhase" ];
+    buildInputs = [ jre ];
 
-  installPhase = ''
-    jar=$out/share/alloy/alloy${version}.jar
+    phases = [ "installPhase" ];
```
Comment (Mic92) [inline pkgs/development/tools/alloy/default.nix]: I would rather not override these phases as it gives users to apply custom modifications when overriding the package.

File: pkgs/development/tools/alloy/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/development/tools/alloy/default.nix]: `makeWrapper` goes to `nativeBuildInputs`

File: pkgs/development/tools/alloy/default.nix [hunk 3]
```diff
@@ -1,58 +1,78 @@
...
+      exec = name;
+      icon = name;
+      desktopName = "Alloy ${major}";
+      genericName = "Relational modelling tool";
+      comment = meta.description;
+      categories = "Development;IDE;Education;";
+    };
 
-  phases = [ "installPhase" ];
+    buildInputs = [ jre ];
```
Comment (Mic92) [inline pkgs/development/tools/alloy/default.nix]: Not really used for building. In the wrapper the full path is specified already.

**Comment by Mic92** [comment]
Please remember to remove alloy4, once alloy5 becomes the stable version.

**Comment by deliciouslytyped** [comment]
Thanks for packaging this!

**Pull request merged**